packages/shared/src/utils/file.utils.ts

```
@@ -0,0 +1,65 @@
+
+  // Split filename into name and extension
+  const lastDotIndex = sanitized.lastIndexOf('.');
+  let name = lastDotIndex > 0 ? sanitized.substring(0, lastDotIndex) : sanitized;
+  let extension = lastDotIndex > 0 ? sanitized.substring(lastDotIndex) : '';
+
+  // Remove path traversal patterns
+  name = name.replace(/\.{2,}/g, '_'); // Replace ".." and "..."
+  extension = extension.replace(/\.{2,}/g, '_');
+
+  // Remove control characters and other dangerous characters
+  const dangerousChars = /[\x00-\x1f\x80-\x9f<>:"/\\|?*]/g;
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)